src/request/sign-multisig-transaction/SignMultisigTransactionApi.js

```diff
@@ -0,0 +1,284 @@
+            )) {
+                throw new Errors.InvalidRequestError(
+                    'Invalid secrets.encrypted: all encrypted secrets must be the same length',
+                );
+            }
+            // Validate keyParams
+            if (!object.secrets.keyParams) {
+                throw new Errors.InvalidRequestError('Missing secrets.keyParams');
+            }
+            const keyParams = object.secrets.keyParams;
+            if (!('kdf' in keyParams) || !('iterations' in keyParams) || !('keySize' in keyParams)) {
+                throw new Errors.InvalidRequestError('Invalid secrets.keyParams: missing properties');
+            }
+            if (!CONFIG.RSA_SUPPORTED_KDF_FUNCTIONS.includes(keyParams.kdf)) {
+                throw new Errors.InvalidRequestError(`Unsupported keyParams KDF function: ${keyParams.kdf}`);
+            }
+            if (!CONFIG.RSA_SUPPORTED_KDF_ITERATIONS[keyParams.kdf].includes(keyParams.iterations)) {
+                throw new Errors.InvalidRequestError(`Unsupported keyParams KDF iterations: ${keyParams.iterations}`);
+            }
+            if (keyParams.keySize !== rsaKeySize) {
+                throw new Errors.InvalidRequestError(`Wrong keyParams key size: ${keyParams.keySize}`);
+            }
+            secrets = {
+                encrypted: object.secrets.encrypted,
+                keyParams,
+            };
+        } else {
+            throw new Errors.InvalidRequestError('Invalid secrets format');
+        }
+
+        const userName = this.parseLabel(object.userName, true, 'userName');
+
+        return {
+            publicKeys,
+            signers,
+            secrets,
+            userName,
+        };
+    }
+
+    /**
+     * @param {Nimiq.Transaction} transaction
+     * @param {MultisigConfig} multisig
+     */
+    verifyMultisigAddress(transaction, multisig) {
+        const multisigAddress = Nimiq.Address.fromPublicKeys(
+            multisig.publicKeys,
+            multisig.signers.length,
+        );
+
+        if (transaction.senderType === Nimiq.AccountType.Basic) {
+            if (!transaction.sender.equals(multisigAddress)) {
+                throw new Errors.InvalidRequestError(
+                    'Transaction sender does not match calculated multisig address',
+                );
+            }
+        } else if (transaction.senderType === Nimiq.AccountType.Vesting) {
+            // Cannot verify vesting contract address
+        } else if (transaction.recipientType === Nimiq.AccountType.Basic) {
+            if (!transaction.recipient.equals(multisigAddress)) {
+                throw new Errors.InvalidRequestError(
+                    'Transaction recipient does not match calculated multisig address',
+                );
+            }
+        } else {
+            throw new Errors.InvalidRequestError(
+                'The multisig account must either be the sender or the recipient of the transaction',
+            );
+        }
```

Comment:
More concise and with documentation:
```suggestion
        if (!(transaction.senderType === Nimiq.AccountType.Basic && transaction.sender.equals(multisigAddress))
            // Vesting contracts owned by the multisig are allowed to send anywhere. The contract address can't be
            // verified here, it can however be assumed to be owned by the multisig, as only the owner can sign a valid
            // vesting withdrawal transaction.
            && transaction.senderType !== Nimiq.AccountType.Vesting
            // For other transactions which are not from a basic account, i.e. from a contract (other than a vesting
            // contract), we enforce the multisig to be the recipient. E.g. unstaking, HTLC refund, HTLC withdrawal etc.
            // are only allowed to the multisig address.
            && !(transaction.recipientType === Nimiq.AccountType.Basic && transaction.recipient.equals(multisigAddress))
        ) {
            throw new Errors.InvalidRequestError(
                'The transaction must either be send from or to the multisig, or from a vesting contract owned by it',
            );
        }
```

Reply:
I do not agree with the more concise version, its not understandable. The documentation can also be added to the more verbose layout.

Reply:
I think the more concise version is actually better to understand, which is why I changed it. It clearly lists the acceptable cases, and does not mix acceptable and unacceptable cases.